lochness/mindlamp/__init__.py

```diff
@@ -99,49 +101,71 @@ def sync(Lochness: 'lochness.config',
         time_utc_24_ts = time.mktime(time_utc_24.timetuple()) * 1000
 
+
         # date string to be used in the file name
         date_str = time_utc_00.strftime("%Y_%m_%d")
 
-        # Extra information for future version
-        # study_id, study_name = get_study_lamp(LAMP)
-        # subject_ids = get_participants_lamp(LAMP, study_id)
-        subject_id = subject.mindlamp[f'mindlamp.{subject.study}'][0]
         logger.debug(f'Mindlamp {subject_id} {date_str} data pull - start')
 
-        # pull data from mindlamp
-        begin = time.time()
-        activity_dicts = get_activity_events_lamp(
-                LAMP, subject_id,
-                from_ts=time_utc_00_ts, to_ts=time_utc_24_ts)
-
-        sensor_dicts = get_sensor_events_lamp(
-                LAMP, subject_id,
-                from_ts=time_utc_00_ts, to_ts=time_utc_24_ts)
-        end = time.time()
-        logger.debug(f'Mindlamp {subject_id} {date_str} data pull - complete')
-
-        # set destination folder
-        dst_folder = tree.get('mindlamp',
-                              subject.protected_folder,
-                              processed=False,
-                              BIDS=Lochness['BIDS'])
-
-        # store both data types
-        for data_name, data_dict in zip(['activity', 'sensor'],
-                                        [activity_dicts, sensor_dicts]):
-
+        # for both type of data
+        for data_name in ['activity', 'sensor']:
             dst = os.path.join(
                     dst_folder,
                     f'{subject_id}_{subject.study}_{data_name}_'
                     f'{date_str}.json')
 
+            function_to_execute = get_activity_events_lamp \
+                if data_name == 'activity' else get_sensor_events_lamp
+
+            # confirm it's a new file
+            if Path(dst).is_file():
+                with open(dst, 'r') as existing_file:
+                    # [-1] at the end -> last timepoint data
+                    existing_data = json.load(existing_file)[-1]
+
+                new_data = function_to_execute(
+                    LAMP, subject_id,
+                    from_ts=existing_data['timestamp'],
+                    to_ts=existing_data['timestamp'])[0]
+
+                existing_data_json = json.dumps(existing_data, sort_keys=True)
+                new_data_json = json.dumps(new_data, sort_keys=True)
+
+                if existing_data_json == new_data_json:
+                    logger.debug(f'{data_name} data exist for {date_str} '
+                                 '- not downloading')
+                    continue
+
+            # pull data
+            begin = time.time()
+            data_dict = function_to_execute(
+                    LAMP, subject_id,
+                    from_ts=time_utc_00_ts, to_ts=time_utc_24_ts)
+            end = time.time()
+            logger.debug(f'Mindlamp {subject_id} {date_str} {data_name}'
+                         'data pull - complete ({end - begin})')
```

Comment:
I think we want `'... in ({end - begin}) seconds'`

Reply:
3826168